gcloud/datastore/__init__.py

```diff
@@ -170,4 +182,7 @@ def allocate_ids(incomplete_key, num_ids):
     :rtype: list of :class:`gcloud.datastore.key.Key`
     :returns: The (complete) keys allocated with `incomplete_key` as root.
     """
-    return _require_dataset().allocate_ids(incomplete_key, num_ids)
+    dataset = _require_dataset()
+    connection = _require_connection()
+    return connection_module.allocate_ids(incomplete_key, num_ids,
```

Comment:
Yeah it's a sacrifice to the lint gods and one that is possibly good. I mentioned on the other commit:

> The rename part was a sacrifice to the lint gods since the global `connection`
> gets overwritten in a function signature as a local.